src/bitcoin-cli.cpp

```diff
@@ -1024,6 +1025,8 @@ __declspec(dllexport) int main(int argc, char* argv[])
 int main(int argc, char* argv[])
 {
 #endif
+    EnableSyscallSandbox(SyscallSandboxPolicy::INITIALIZATION);
+    DisableFurtherSyscallSandboxRestrictions();
```

Comment:
The `DisableFurtherSyscallSandboxRestrictions` is (or rather: was) used to make any further/future `EnableSyscallSandbox` calls that be into no-ops.

In other words to "lock" the sandbox to the sandbox policy currently loaded. (Or disable the sandbox if no sandbox is loaded: in other words a call to `DisableFurtherSyscallSandboxRestrictions` without any preceding `EnableSyscallSandbox`.)

The quoted code above for example loads the `SyscallSandboxPolicy::INITIALIZATION` policy on `bitcoin-cli` startup and then locks that policy so that no matter what functions that `bitcoin-cli` may call that uses `EnableSyscallSandbox` we're guaranteed to run under `SyscallSandboxPolicy::INITIALIZATION`.

`SyscallSandboxPolicy::INITIALIZATION` is the super-set of all sandbox policies in Bitcoin Core, so the code above basically says "limit me to the syscall set we expect to be used in (any part of) Bitcoin Core, but make sure no further syscall sandbox restrictions are done in code I may call" (in other words: ignore any upcoming `EnableSyscallSandbox` calls).

The above is how it used to be, and I agree that was a bit confusing, so I've now updated things:

Instead of enabling the sandbox on first call to `void EnableSyscallSandbox(policy)`  I've now added a separate setup function `bool SetupSyscallSandbox(…)` (called at most once per binary) to enable the sandbox, and added a separate `void SetSyscallSandboxPolicy(policy)` to load a specific policy.

In other words the sandbox is opt-in via `bool SetupSyscallSandbox(…)`.

A call to `SetupSyscallSandbox(…)` is made in `bitcoind` if `-sandbox=<mode>` is used.

In this PR I'm now limiting the syscall sandbox support to the `bitcoind` binary to keep things focused. I suggest we wait with the other binaries to keep things focused. We can gradually enable the sandbox for other binaries over time in future PR:s as we see fit when we've gained some experience with the experimental `bitcoind` syscall sandbox support. Sounds reasonable? :)